src/topology/index.js

```diff
@@ -26,6 +26,7 @@ class Topology {
     this._onConnect = handlers.onConnect || noop
     this._onDisconnect = handlers.onDisconnect || noop
 
+    // TODO: can this be a set?
```

Comment:
The issue with making this a Set is that we can't test equality of complex objects. So if we end up with a copy of an existing peer it will get added to the set. The Map gets around this by using the peer id string as the key.